Aristos espionage add on

Azurency opened the pull request:
Update from Astog, you can read more about it here https://github.com/Azurency/CQUI_Community-Edition/issues/234

Azurency committed:
Astog espionage add on

ricanuck commented:
Credit where it is due. Right? It is not an update FROM astog. Nor FROM you.

ricanuck reviewed (changes requested):
Credit has to go where it is due, no matter how much any person may disagree with any other. THAT is how open source works. "Stealing" code without shame and due credit is not acceptable (even if it is a few lines).

ricanuck commented on UI/EspionageSupport.lua:
```diff
@@ -129,7 +129,7 @@ function hasDistrict(city:table, districtType:string)
     local cityDistricts:table = city:GetDistricts();
     for i, district in cityDistricts:Members() do
-        if district:IsComplete() then
+        if district:IsComplete() and not district:IsPillaged() then
```
Insert for due credit:
--ARISTOS: to only show available and valid targets in each city, both for espionage overview and selector

ricanuck commented on UI/PartialScreens/EspionageOverview.lua:
```diff
@@ -421,20 +421,21 @@ function AddDistrictIcon(stackControl:table, city:table, districtType:string)
 
     -- We're manipulating the alpha to hide each element so they maintain their stack positions
-    if hasDistrict then
+    if hasDistrict(city, districtType) then
```
Insert for due credit:
--ARISTOS: make use of the espionagesupport.lua funtion, more efficient and has been fixed to only show valid targets

ricanuck commented on UI/PartialScreens/EspionageOverview.lua:
```diff
@@ -421,20 +421,21 @@ function AddDistrictIcon(stackControl:table, city:table, districtType:string)
     local toolTipString:string = "";
 
     -- Check if we have this district
-    local hasDistrict:boolean = false;
```
Erase all this, no need to leave it block commented.

Azurency committed:
Update with credits

Really sorry I messed up with the credit in the last commit message, I was not paying enough attention.

Azurency commented:
Wow, I'm really sorry, I wrote Astog without even noticing (might be because I was in his repository). Of course I meant you Aristos, I corrected in the code also with your credits.

ricanuck commented:
Thanks.

Azurency commented:
We've merged this changes in CQUI.
